includes/class-wsuwp-people-rest-api.php

```diff
@@ -52,6 +58,108 @@ public function show_people_in_rest() {
 	}
 
 	/**
+	 * Authenticates a cross domain request to edit a person record.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @global wp $wp
+	 *
+	 * @param WP_Error|mixed $result Error from another authentication handler. null if we
+	 *                               should handle it, another value if not.
+	 *
+	 * @return WP_Error|null|bool WP_Error if authentication error.
+	 *                            null if authentication wasn't used.
+	 *                            true if authentication succeeded.
+	 */
+	public function rest_verify_authentication( $result ) {
+		global $wp;
+
+		if ( ! empty( $result ) ) {
+			return $result;
+		}
+
+		// Only perform custom authentication on the people endpoint.
+		if ( 0 !== strpos( $wp->request, 'wp-json/wp/v2/people' ) ) {
+			return null;
+		}
+
+		// Determine if there is a nonce.
+		$nonce = null;
+
+		if ( isset( $_REQUEST['_wpnonce'] ) ) { // @codingStandardsIgnoreLine
+			$nonce = $_REQUEST['_wpnonce']; // @codingStandardsIgnoreLine
+		} elseif ( isset( $_SERVER['HTTP_X_WP_NONCE'] ) ) {
+			$nonce = $_SERVER['HTTP_X_WP_NONCE'];
+		}
+
+		$uid = null;
+
+		if ( isset( $_SERVER['HTTP_X_WSUWP_UID'] ) ) {
+			$uid = absint( $_SERVER['HTTP_X_WSUWP_UID'] );
+		}
+
+		$domain = null;
+
+		if ( isset( $_SERVER['HTTP_ORIGIN'] ) ) {
+			$domain = wp_parse_url( $_SERVER['HTTP_ORIGIN'], PHP_URL_HOST );
+		}
+
+		if ( null === $nonce || ! $uid || empty( $domain ) ) {
+			// No nonce at all, so act as if it's an unauthenticated request.
+			wp_set_current_user( 0 );
+			return true;
```

Comment:
`true` is okay here. We explicitly set the user to 0, so even though authentication is successful, no edits will be allowed. A standard `GET` request should still work fine.